bliss/generate.py

```diff
@@ -0,0 +1,98 @@
+import os
+import pickle
+from typing import Dict, List, TypedDict
+
+import torch
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+from torch.utils.data import IterableDataset
+from tqdm import tqdm
+
+from bliss.catalog import TileCatalog
+
+
+def generate(cfg: DictConfig):
+    simulator = instantiate(cfg.simulator)
+
+    file_data_capacity = cfg.cached_simulator.file_data_capacity
+    cached_data_path = cfg.cached_simulator.cached_data_path
+
+    # largest `batch_size` multiple <= `file_data_capacity`
+    file_data_size = (
+        file_data_capacity // simulator.image_prior.batch_size
+    ) * simulator.image_prior.batch_size
```

Comment:
Would you write this a bit more simply, e.g., 
```
    bs = simulator.image_prior.batch_size
    file_data_size = (file_data_capacity // bs) * bs
```

Reply:
Done.